src/view/configuration/utils/copyPropertiesIfValueDifferentThanDefault.js

```diff
@@ -9,20 +9,61 @@ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTA
 OF ANY KIND, either express or implied. See the License for the specific language
 governing permissions and limitations under the License.
 */
+
 /**
  * Copies properties from one object to another, but only if the value
  * doesn't match the default value.
  * @param {Object} options
  * @param {Object} options.toObj Object to which properties should be copied.
  * @param {Object} options.fromObj Object from which properties should be copied.
  * @param {Object} options.defaultsObj Default values for each property.
- * @param {Array} options.keys The keys of the properties that should be attempted
- * to be copied.
+ * @param {Array} options.keys The top-level keys of the properties that should be attempted
+ * to be copied. Does not apply to nested keys
  */
-export default ({ toObj, fromObj, defaultsObj, keys }) => {
+export default function copyPropertiesIfValueDifferentThanDefault({
+  toObj,
+  fromObj,
+  defaultsObj,
+  keys
+}) {
   keys.forEach(key => {
-    if (fromObj[key] !== defaultsObj[key]) {
-      toObj[key] = fromObj[key];
+    const fromValue = fromObj[key];
+    const defaultValue = defaultsObj[key];
+
+    if (Array.isArray(fromValue) && Array.from(defaultValue)) {
+      if (fromValue.length !== defaultValue.length) {
+        toObj[key] = fromValue;
+        return;
+      }
+      // does not support arrays of objects
+      const sortedFromValue = fromValue.sort();
+      const sortedDefaultValue = defaultValue.sort();
```

Comment:
sort mutates the original array. (i.e. sortedFromValue and fromValue are the same). You'd have to do fromValue.slice(0).sort() if you didn't want it to mutate.